File: python/maplestory_openapi/api/dto/character/character_stat.py
from datetime import datetime
from pydantic import BaseModel


class CharacterStatInfo(BaseModel):
    """캐릭터 스탯 상세 정보

    stat_name(str): 스탯 이름
    stat_value(str): 스탯 값
    """
    stat_name: str
    stat_value: str


class CharacterStat(BaseModel):
    """캐릭터 스탯 정보

    date(datetime): 조회 기준일
    character_class(str): 직업
    final_stat(list[CharacterStatInfo]): 현재 스탯 정보
    remain_ap(int): 잔여 AP
    """
    date: datetime | None
    character_class: str
    final_stat: list[CharacterStatInfo]
    remain_ap: int

File: js/src/maplestory/api/tms/dto/character/characterStat.ts
import * as base from '../../../common/dto/character/characterStat';
import {
  CharacterFinalStatBody,
  CharacterStatBody,
} from '../../response/character/characterStatBody';

/**
 * 角色綜合能力值資訊
 */
export class CharacterStatDto extends base.CharacterStatDto {
  /**
   * 要搜尋的日期 (TST，每日資料中的小時與分鐘將顯示為 0)
   */
  public override date: Date | null;

  /**
   * 角色職業
   */
  public override characterClass: string | null;

  /**
   * 當前能力值資訊
   */
  public override finalStat: CharacterFinalStatDto[];

  /**
   * 剩餘 AP
   */
  public override remainAP: number | null;

  constructor(obj: CharacterStatBody) {
    super();

    const { date, character_class, final_stat, remain_ap } = obj;

    this.date = date ? new Date(date) : null;
    this.characterClass = character_class;
    this.finalStat = final_stat.map((stat) => new CharacterFinalStatDto(stat));
    this.remainAP = remain_ap;
  }
}

/**
 * 當前能力值資訊
 */
export class CharacterFinalStatDto extends base.CharacterFinalStatDto {
  /**
   * 能力值名稱
   */
  public override statName: string;

  /**
   * 能力值數值
   */
  public override statValue: string;

  constructor(obj: CharacterFinalStatBody) {
    super();

    const { stat_name, stat_value } = obj;

    this.statName = stat_name;
    this.statValue = stat_value;
  }
}
